components/Form/index.js

```diff
@@ -0,0 +1,328 @@
+import styled from "styled-components";
+
+export default function Form({
+  handleAddPlant,
+  plant,
+  buttonText,
+  handleToggleModal,
+  handleEditPlant,
+}) {
+  function handleSubmitAddPlant(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    const selectedSeasons = formData.getAll("fertiliserSeason");
+    if (selectedSeasons.length === 0) {
+      alert("Please select at least one season.");
+      return;
+    }
+    data.fertiliserSeason = selectedSeasons;
+
+    { buttonText === "Edit" && handleEditPlant(data, plant.id) }
+
+    { buttonText === "Add" && handleAddPlant(data) }
+
+    event.target.reset();
+}
+  return (
+    <StyledSection>
+      <StyledH2>
+        {buttonText === "Edit" ? `Edit ${plant.name}` : "Add plant"}
+      </StyledH2>
+      <form onSubmit={handleSubmitAddPlant}>
+        <StyledFieldset>
+          <label htmlFor="plantName">Plant name:</label>
+          <StyledInput
+            id="plantName"
+            name="name"
+            type="text"
+            placeholder="e.g. Monstera"
+            required
+            defaultValue={buttonText === "Edit" ? plant.name : ""}
+          ></StyledInput>
+        </StyledFieldset>
+        <StyledFieldset>
+          <label htmlFor="botanicalPlantName">Botanical plant name:</label>
+          <StyledInput
+            id="botanicalPlantName"
+            name="botanicalName"
+            type="text"
+            placeholder="e.g. Monstera deliciosa"
+            required
+            defaultValue={buttonText === "Edit" ? plant.botanicalName : ""}
+          ></StyledInput>
+        </StyledFieldset>
+        <StyledFieldset>
+          <label htmlFor="plantDescription">Description:</label>
+          <StyledTextarea
+            id="plantDescription"
+            name="description"
+            type="text"
+            placeholder="e.g. Monstera deliciosa, also known as the Swiss cheese plant"
+            defaultValue={buttonText === "Edit" ? plant.description : ""}
+          ></StyledTextarea>
+        </StyledFieldset>
+        {/*-----------------------------------------------------------*/}
+        <StyledFieldsetRadio>
+          <legend> Light needs </legend>
+
+          <StyledRadioInput
+            id="light-full-shade"
+            name="lightNeed"
+            type="radio"
+            value="Full Shade"
+            defaultChecked
+          />
+          <StyledRadiolabel htmlFor="light-full-shade">
+            Full Shade{" "}
+          </StyledRadiolabel>
+
+          <StyledRadioInput
+            id="light-partial-shade"
+            name="lightNeed"
+            type="radio"
+            value="Partial Shade"
+            defaultChecked={
+              buttonText === "Edit" && plant.lightNeed == "Partial Shade"
+                ? true
+                : false
+            }
```

Comment:
Since `buttonText === "Edit" && plant.lightNeed == "Partial Shade"` can only be true/false you don't have to write it explicitly. Simply write `defaultCheckerd={buttonText === "Edit" && plant.lightNeed == "Partial Shade"}`. The same applies to all following inputs, radios, ...

Reply:
this is great, thanks!